api/src/groups/api.py

```diff
@@ -5,527 +5,507 @@
+    except ValidationError as err:
+        return jsonify(err.messages), 422
+
+    if 'active' not in valid_group:
+        valid_group['active'] = True
+
+    new_group = Group(**valid_group)
+
+    db.session.add(new_group)
+
+    try:
+        db.session.commit()
+    # when the foreign key given is invalid
+    except IntegrityError:
+        return jsonify('Payload contains an invalid foreign key'), 404
+
+    return jsonify(group_schema.dump(new_group)), 201
 
 
 @groups.route('/groups')
 def read_all_groups():
     query = db.session.query(Group)
-    return_group = request.args.get('return_group')
-    if return_group == 'inactive':
-        query = query.filter_by(active=False)
-    elif return_group in ('all', 'both'):
-        pass  # Don't filter
-    else:
-        query = query.filter_by(active=True)
-    query = query.all()
-
-    for group in query:
-        group.member_list = group.members
-        group.manager_info = group.manager
-        group.manager_info.person = group.manager.person
-    return jsonify(group_schema.dump(query, many=True))
+    try:
+        groups = get_all_queried_entities(query, request.args)
+    except QueryArgumentError as e:
+        return jsonify(e.message), e.code
+    group_schema = GroupSchema()
+    return jsonify(group_schema.dump(groups, many=True)), 200
 
 
-@groups.route('/groups/<group_id>')
+@groups.route('/groups/<int:group_id>')
 @jwt_required
 def read_one_group(group_id):
-    result = db.session.query(Group).filter_by(id=group_id).first()
-    if result is None:
-        return jsonify(msg="Group not found"), 404
-    return group_dump(result), 200
-
-
-@groups.route('/find_group/<group_name>/<manager>')
-@jwt_required
-def find_group(group_name=None, manager=None):
-    matching_group_count = db.session.query(Group).filter_by(name=group_name, manager_id=manager).count()
-    return jsonify(matching_group_count), 200
-
+    group = db.session.query(Group).filter_by(id=group_id).first()
+    if group is None:
+        return jsonify(f"Group with id {group_id} does not exist"), 404
+    return jsonify(group_schema.dump(group)), 200
 
-@groups.route('/groups/<group_id>', methods=['PATCH'])
+@groups.route('/groups/<int:group_id>', methods=['PATCH'])
 @jwt_required
 def update_group(group_id):
-    # fetch the optional 'person_ids' field in the request object
-    update_person_ids = []
-    if 'person_ids' in request.json.keys():
-        update_person_ids = request.json['person_ids']
-        del request.json['person_ids']
-
+    group_schema = GroupSchema()
     try:
-        valid_group = group_schema.load(request.json)
+        valid_attributes = group_schema.load(request.json, partial=True)
     except ValidationError as err:
         return jsonify(err.messages), 422
 
-    group = db.session.query(Group).filter_by(id=group_id).first()
-    if group is None:
-        return jsonify(msg="Group not found"), 404
-
-    new_manager_id = None
```

Comment:
Is this code going away entirely? If so, is that the case because of the database changes?

Reply:
I think the `modify_entity` function will return a 404 error in case the group specified does not exist. Not sure if that's the best practice, but I noticed that this `modify_entity` function is used throughout the `event` module and thought it's probably better to reuse something that's already there... We should definitely change the interface to be more transparent, if needed.

Reply:
Agreed.